src/fetch.js

```diff
@@ -136,13 +152,133 @@ function normalizeUrl(input) {
   throw new TypeError(`Unsupported protocol: '${protocol}'`);
 }
 
+const cacheConfig = {
+  'default': { useCache: true, conditionalRequest: true, updateCache: true },
+  'no-store': {},
+  'reload': { updateCache: true },
+  'no-cache': { conditionalRequest: true, updateCache: true },
+  'force-cache': { useCache: true, allowStale: true, updateCache: true },
+  'only-if-cached': { useCache: true, allowStale: true, skipNet: true },
+};
+
+/**
+ * @type {{[key:string]:Promise<Response>}}
+ */
+const concurrentRequests = {};
+
+/**
+ * Share concurrent requests/responses for the same resource.
+ * @param {string} cacheKey 
+ * @param {()=>Promise<Response>} next
+ * @returns {Promise<Response>}
+ */
+async function checkConcurrent(cacheKey, next) {
+  let current = concurrentRequests[cacheKey];
+  if (!current) {
+    current = next();
+    concurrentRequests[cacheKey] = current;
+    setTimeout(() => {
```

Comment:
what does this `setTimeout` do?  I see the code, but don't understand what it's doing.

Reply:
Removes the current request after 1000 ms ?
Since it invokes `unref`, if there is no other activity keeping the event loop running, the process may exit before the Timeout object's callback is invoked.

Reply:
Exactly.  I added some comments to help.